DigitalOffice/Services/ProjectService/Repositories/ProjectRepository.cs

```diff
@@ -40,5 +41,24 @@ public Guid CreateNewProject(DbProject newProject)
 
             return newProject.Id;
         }
+
+        public Guid EditProjectById(Guid projectId, DbProject dbProject)
+        {
+            var projectToEdit = dbContext.Projects
+                .AsNoTracking()
+                .SingleOrDefault(p => p.Id == projectId);
+
+            if (projectToEdit == null)
+            {
+                throw new NullReferenceException("Project with this Id does not exist");
+            }
+
+            dbProject.Id = projectId;
+
+            dbContext.Projects.Update(dbProject);
```

Comment:
Update method documentation says : "For entity types with generated keys if an entity has its primary key value set then it will be tracked in the Modified state. If the primary key value is not set then it will be tracked in the Added state. This helps ensure new entities will be inserted, while existing entities will be updated. An entity is considered to have its primary key value set if the primary key property is set to anything other than the CLR default for the property type"

Well, if I understood clearly, dbProject should have not default projectId

Reply:
Umm, projectId = Guid from the query. This Guid is passed to the entity and update works correctly.

Reply:
Sorry, my mistake. Then what's the point in separating the project and the id?

Reply:
I think there's none. So I'll update the code.